### PR TITLE
Fix absolute mains

### PR DESCRIPTION
**Pull request by fendor**
Adds failing test-case.

### CODE REVIEW

**Comment by fendor** [comment]
@maksbotan Could you check if this fixes your problem, too?

**Comment by maksbotan** [comment]
Looks like it broke something :(
```
    space stack:               FAIL (1.56s)
      Finding Cradle for: /tmp/dist-test.rmDk/hie-bios-0.5.0/tests/projects/space stack/B.hs
      Loading Cradle: Just "/tmp/dist-test.rmDk/hie-bios-0.5.0/tests/projects/space stack/hie.yaml"
      Initialise Flags                                               (1.56s)
      Initial module load
        can't find file: /tmp/dist-test.rmDk/hie-bios-0.5.0/tests/projects/space/B.hs
```

**Comment by maksbotan** [comment]
On the other hand, it does fix my issue, thanks!

I like you code more, I just didn't think about looking into what `Target` is :)

**Comment by fendor** [comment]
@maksbotan It actually uncovered an existing bug #206 which needs to be fixed.

File: src/HIE/Bios/Environment.hs [hunk 1]
```diff
@@ -47,9 +47,20 @@ initSession  ComponentOptions {..} = do
...
+    return targets'
+
+----------------------------------------------------------------
+
+normaliseTargets :: FilePath -> [G.Target] -> [G.Target]
+normaliseTargets wdir = map (\target -> target {G.targetId = normaliseTargetId wdir (G.targetId target)})
+
+normaliseTargetId :: FilePath -> G.TargetId -> G.TargetId
+normaliseTargetId wdir (G.TargetFile fp phase) = G.TargetFile (prependIfRelative wdir fp) phase
+normaliseTargetId _ tid = tid
```
Comment (fendor) [inline src/HIE/Bios/Environment.hs:63]: This should probably be named 'makeTargetsAbsolute' to mirror the naming of `makeDynFlagsAbsolute`.
@jneira, @bubba opinions?

Comment (bubba) [reply src/HIE/Bios/Environment.hs:63]: Yeah normalise to me means something more like `canonicalizePath`

**Pull request merged**